src/actions/actions.js

```diff
@@ -0,0 +1,53 @@
+import axios from "axios";
+import { unionBy } from "lodash/array";
+import { getLinkifiedHTML } from "./util";
+
+const notice_urls = ["acad_ug", "acad_pg", "bcrth", "public"];
+const veritas_url = "https://hermes.mykgp.com/";
+
+var updateNotices = notices => ({
+  type: "UPDATE_NOTICES",
+  value: notices
+});
+
+var setFetchingNotices = value => ({
+  type: "SET_FETCHING_NOTICES",
+  value: value
+});
+
+export var fetchNotices = (noticeType, firstPageOnly = false) => (
+  dispatch,
+  getState
+) => {
+  // getState() returns mutable, make sure to create copy
+  var notices = { ...getState().noticesData.notices };
+  var currentTypeNotices = notices[noticeType];
+  const { allFetched, nextPage } = currentTypeNotices;
+  if (!allFetched) {
+    var url = `${veritas_url}${notice_urls[noticeType]}/`;
+    if (nextPage && !firstPageOnly) {
+      url = `${url}page/${nextPage}`;
+    }
+    dispatch(setFetchingNotices(true));
+    axios
+      .get(url)
+      .then(response => {
+        const { data, nextPage } = currentTypeNotices;
```

Comment:
`nextPage` is getting redefined here (there's one in the outer scope too). Could possibly rename this.